ipad/ViewControllers/Shift/Cells/Shared cells/Button/FormButtonCollectionViewCell.swift

```diff
@@ -11,15 +11,61 @@ import UIKit
 class FormButtonCollectionViewCell: UICollectionViewCell, Theme {
 
     @IBOutlet weak var button: UIButton!
+    @IBOutlet weak var dryStorageSwitch: UISwitch?
+    @IBOutlet weak var switchLabel: UILabel?
+    
+    struct Config {
+        var status: Bool = false
+        var isPreviousJourney: Bool = false
+        var displaySwitch: Bool = false
+    }
+    
+    
     var completion: (()-> Void)?
+    @objc weak var target: NSObject?
+    @objc var _selector: Selector?
     
     @IBAction func buttonAction(_ sender: UIButton) {
+        let _ = self.target?.perform(_selector, with: [:])
         guard let onClick = self.completion else {return}
         return onClick()
     }
     
-    func setup(with title: String, isEnabled: Bool, onClick: @escaping ()-> Void) {
+    func set(status: Bool) {
+        self.button?.isEnabled = !status
+    }
+    
+    func setupConfig(config: Config) {
```

Comment:
Looks good but maybe this can be set to private to not confuse the dev on which setup function to use

Reply:
DONE